### PR TITLE
Bug 1467877 - add jsone_templates that parses multiple YAML docs

### PR DESCRIPTION
**Pull request by djmitche**
So the process is:
 * ingest bytestream
 * produce go-yaml YAML data structures
 * encode to YAML
 * ghodss decode
   * decode to go-YAML data structures
   * convert to JSON data structures
   * encode in JSON
   * decode to JSON data structures
 * encode to JSON or YAML depending on the `format` argument

But, it works!

### CODE REVIEW

**Comment by djmitche** [comment]
P.S. The thought crossed my mind to just shell out to Python :)

File: jsoneprovider/data_source_jsone_template.go [hunk 1]
```diff
@@ -51,44 +61,99 @@ func dataSourceJsoneTemplate() *schema.Resource {
...
 	if err != nil {
 		return err
 	}
 
-	var m []byte
-	if format == "json" {
-		m, err = json.Marshal(result)
-	} else {
-		m, err = yaml.Marshal(result)
+	fmt.Printf("%#v\n", rendered)
```
Comment (imbstack) [inline jsoneprovider/data_source_jsone_template.go]: Stray debug?

File: yaml/decoder_test.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+	"fmt"
+	"io"
+	"testing"
+
+	"github.com/renstrom/dedent"
+	"github.com/stretchr/testify/require"
+)
+
+func TestDecoder(t *testing.T) {
+	doc := dedent.Dedent(`
```
Comment (imbstack) [inline yaml/decoder_test.go:14]: Oh, `dedent` is awesome. I didn't know about it.

**Review by imbstack — APPROVED** [review]
It works and that is good enough for me!

File: yaml/decoder_test.go (hunk 2 above)
Comment (djmitche) [reply yaml/decoder_test.go:14]: I use it all the time in Python :)

**Pull request merged**